components/kyma-environment-broker/internal/storage/driver/postsql/instance.go

```diff
@@ -330,7 +330,10 @@ func (s *Instance) GetByID(instanceID string) (*internal.Instance, error) {
 		}
 		return nil, err
 	}
+	// note for migration, instance has more up-to-date version of migration state than a previous operation
+	migrationTriggered := instance.InstanceDetails.SCMigrationTriggered
 	instance.InstanceDetails = lastOp.InstanceDetails
+	instance.InstanceDetails.SCMigrationTriggered = migrationTriggered
```

Comment:
Instance Details is a struct which describes some created/computed/gathered data for given instance. Because of some historical reasons (now I would implement it a different way) those data are stored in the operations table, but logically it belongs to the instance. That's why if you call a storage to get the instance - you will get also instance details. I think you don't need to do such hack.